FeedStoreChallenge/ManagedCache.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  ManagedCache.swift
+//  FeedStoreChallenge
+//
+//  Created by Shinzan Takata on 2021/04/29.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import CoreData
+
+@objc(ManagedCache)
+class ManagedCache: NSManagedObject {
+	@NSManaged var timestamp: Date
+	@NSManaged var feed: NSOrderedSet
+}
+
+extension ManagedCache {
+	static func find(in context: NSManagedObjectContext) throws -> ManagedCache? {
+		let request = NSFetchRequest<ManagedCache>(entityName: entity().name!)
+		request.returnsObjectsAsFaults = false
+		return try context.fetch(request).first
+	}
+
+	static func newUniqueInstance(in context: NSManagedObjectContext) throws -> ManagedCache {
+		try deleteCache(in: context)
+		return ManagedCache(context: context)
+	}
+
+	private static func deleteCache(in context: NSManagedObjectContext) throws {
+		try find(in: context).map(context.delete).map(context.save)
```

Comment:
No need to save a context after each operation. It even can be harmful in some cases. Imagine that `CoreDataFeedStore` is called, then previously stored cache is removed successfully and the context is saved. Then the program tries to insert a new instance of `ManagedCache` and fails (out of memory or some disk IO failure). The program would catch the error and rollback the context to the last save point which is after the cache was deleted. Thus it will end up without both new and cached feed.

Please remove `context.save`

```suggestion
		try find(in: context).map(context.delete)
```

Reply:
Thank you. I fixed it.